### ISSUE TITLE
Order of variables

### ISSUE THREAD

**Issue by krlmlr** [tidyverse/dplyr#3481]
This is a meta-issue to describe possible changes in the order of variables after certain operations. Fixing this would be a breaking change, but worthwhile in the long run to improve consistency across verbs.

- [ ] `group_by_all(.add = TRUE)` shouldn't change the order of existing group variables, https://github.com/tidyverse/dplyr/pull/3409#discussion_r175774842
- [ ] Group variables are put first after `transmute_all()`, https://github.com/tidyverse/dplyr/pull/3409#discussion_r179596117

**Comment by hadley** [comment]
Another one:

``` r
library(dplyr, warn.conflicts = FALSE)

df <- tibble(g = c(1, 2), x = c(1, 2))

df %>% 
  group_by(g) %>% 
  distinct(x)
#> # A tibble: 2 x 2
#> # Groups:   g [2]
#>       x     g
#>   <dbl> <dbl>
#> 1     1     1
#> 2     2     2
```

<sup>Created on 2019-12-12 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by hadley** [comment]
The problem with `transmute_all()` appears to have its root cause in `transmute()`, and is now https://github.com/tidyverse/dplyr/issues/4693.

I couldn't reproduce the problem with `group_by_all()`:

``` r
library(dplyr, warn.conflicts = FALSE)

df <- tibble(x = 1, g = 2)

df %>% 
  group_by(g) %>% 
  group_by_all(.add = TRUE) %>% 
  group_vars()
#> [1] "g" "x"

df %>% 
  group_by(g, x) %>% 
  group_by_all(.add = TRUE) %>% 
  group_vars()
#> [1] "g" "x"

df %>% 
  group_by(x, g) %>% 
  group_by_all(.add = TRUE) %>% 
  group_vars()
#> [1] "x" "g"
```

<sup>Created on 2020-01-05 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

The problem with `distinct()` will be fixed in #4691 (and appears to be an accidental regression)
